crates/task-impls/src/transactions.rs

```diff
@@ -164,11 +166,47 @@ impl<
         None
     }
 
+    /// Get last known builder commitment from consensus.
+    async fn latest_known_builder_commitment(&self) -> BuilderCommitment {
+        let consensus = self.consensus.read().await;
+
+        let mut prev_view = TYPES::Time::new(self.cur_view.saturating_sub(1));
+
+        // Search through all previous views...
```

Comment:
Does this mean searching through all previous decided views? If yes, isn't that inefficient?

Reply:
We'll stop almost immediately in practice, I don't think it's a huge concern.
Ideally we shoild explicitly store last known builder commitment, you are right

Reply:
I believe it searches until it hits a decided view, and breaks out of the loop (and the rest of the function) at that point.

We could do a binary search or something else more asymptotically worst-case efficient. But I don't think we'll ever be that far from a decided leaf, so I feel like a simple linear backwards search makes more sense.

Reply:
Thanks! let's merge this.